src/tree/branch.rs

```diff
@@ -88,10 +86,16 @@ impl<const DEPTH: usize> PoseidonBranch<DEPTH> {
 
     /// Represents the root for a given path of an opening over a subtree
     pub fn root(&self) -> &BlsScalar {
-        self.path
-            .last()
-            .map(|l| l.deref())
-            .unwrap_or(&Self::NULL_ROOT)
+        &self.root
+    }
+
+    /// Returns a mutable reference to the level with the given `index` in the
+    /// path.
+    ///
+    /// NOTE: This should **never** be exposed to the consumer of the crate.
+    #[allow(dead_code)]
+    pub(crate) fn level_mut(&mut self, index: usize) -> &mut PoseidonLevel {
+        &mut self.path[index]
```

Comment:
At first I was a bit puzzled by the comment and the `#[allow]`, then I noticed this was used in a *unit test* but on a different module.

I would revert the change on the visibility as it was before. I'm not sure why it was changed.
Having  the `path` field directly `pub(crate)` feels like a better ergonomic: beside the extra method only used in the tests, there is no dead code and an extra comment to explain the additional method.

Reply:
I think both things are pretty unseemly, but I don't have a strong opinion of one vs. the other. I'll change it back.

Reply:
Done